sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/storage/GcsResourceId.java

```diff
@@ -91,6 +91,17 @@ public String getScheme() {
     return GcsFileSystemRegistrar.GCS_SCHEME;
   }
 
+  @Override
+  public String getFilename() {
+    if (gcsPath.getNameCount() < 1) {
+      return null;
+    } else if (gcsPath.getNameCount() == 1) {
+      return gcsPath.getBucket();
+    } else {
+      return gcsPath.getFileName().toString();
+    }
```

Comment:
I feel like this should return `null` for `<= 1` and never return bucket -- `gs://bucket/` is equivalent to `/` , right?